pallets/proposals/src/impls.rs

```diff
@@ -144,7 +145,9 @@ impl<T: Config> Pallet<T> {
         project.agreement_hash = agreement_hash;
 
         // Add project to list
-        <Projects<T>>::insert(project_key, project);
+        Projects::<T>::mutate_exists(project_key, |p| {
```

Comment:
Maybe just use try_mutate as Hector suggested, youre never mutating to None also so may aswell.
Also do the mutation inside the mutate?

```suggestion
        Projects::<T>::try_mutate(project_key, |p| { if let Some(project) = p {p.milestones = milestones, ...})?;
```